tests/performance/density-kwok.go

```diff
@@ -0,0 +1,322 @@
+				Expect(err).NotTo(HaveOccurred(), "failed to delete fake nodes")
+			})
+		})
+	})
+})
+
+func createFakeNodesWithKwok(kubevirtClient kubecli.KubevirtClient) {
+	for i := 1; i <= nodeCount; i++ {
+		nodeName := fmt.Sprintf("kwok-node-%d", i)
+		node := newFakeNode(nodeName)
+		_, err := kubevirtClient.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+		if err != nil {
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create node %s", nodeName))
+		}
+	}
+}
+
+func newFakeNode(nodeName string) *k8sv1.Node {
+	node := &k8sv1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+			Labels: map[string]string{
+				"beta.kubernetes.io/arch":       "amd64",
+				"beta.kubernetes.io/os":         "linux",
+				"kubernetes.io/arch":            "amd64",
+				"kubernetes.io/hostname":        nodeName,
+				"kubernetes.io/os":              "linux",
+				"kubernetes.io/role":            "agent",
+				"node-role.kubernetes.io/agent": "",
+				"kubevirt.io/schedulable":       "true",
+				"type":                          "kwok",
+			},
+			Annotations: map[string]string{
+				"node.alpha.kubernetes.io/ttl": "0",
+				"kwok.x-k8s.io/node":           "fake",
+			},
+		},
+		Spec: k8sv1.NodeSpec{
+			Taints: []k8sv1.Taint{
+				{
+					Key:    "kwok.x-k8s.io/node",
+					Value:  "fake",
+					Effect: "NoSchedule",
+				},
+				{
+					Key:    "CriticalAddonsOnly",
+					Effect: k8sv1.TaintEffectNoSchedule,
+				},
+			},
+		},
+
+		Status: k8sv1.NodeStatus{
+			Allocatable: k8sv1.ResourceList{
+				k8sv1.ResourceCPU:               resource.MustParse("32"),
+				k8sv1.ResourceMemory:            resource.MustParse("256Gi"),
+				k8sv1.ResourceEphemeralStorage:  resource.MustParse("100Gi"),
+				k8sv1.ResourcePods:              resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("1k"),
+				"devices.kubevirt.io/tun":       resource.MustParse("1k"),
+				"devices.kubevirt.io/vhost-net": resource.MustParse("1k"),
+			},
+			Capacity: k8sv1.ResourceList{
+				k8sv1.ResourceCPU:               resource.MustParse("32"),
+				k8sv1.ResourceMemory:            resource.MustParse("256Gi"),
+				k8sv1.ResourceEphemeralStorage:  resource.MustParse("100Gi"),
+				k8sv1.ResourcePods:              resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("1k"),
+				"devices.kubevirt.io/tun":       resource.MustParse("1k"),
+				"devices.kubevirt.io/vhost-net": resource.MustParse("1k"),
+			},
+		},
+	}
+
+	return node
+}
+
+func createFakeVMIBatchWithKWOK(kubevirtClient kubecli.KubevirtClient) {
+	for i := 1; i <= vmCount; i++ {
+		vmi := newFakeVMISpecWithResources()
+
+		_, err := kubevirtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Is it really an issue not to wait?

Reply:
Since we are creating large number of VMIs, the primary concern here is for the K8s API server to handle multiple requests.
May be we can reduce the sleep time to 10ms or 1ms, in this case.

Reply:
Did you ever try without delay? IMO the K8S API should be able to handle that.

Reply:
Tried it now, API server is able to handle that. I will completely eliminate this sleep time.